functions.php

```diff
@@ -1,34 +1,38 @@
 <?php
+/**
+ * Database query, fetch all
+ *
+ * @return returns all data from the database `characters`
+ */
+function databaseFetchAll() {
+    $db = new PDO('mysql:host=db; dbname=collection_project', 'root', 'password');
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    $query = $db->prepare("SELECT `charname`, `class`, `level`, `strength`, `dexterity`, `constitution`, `intelligence`, `wisdom`, `charisma` FROM `characters`;");
+    $query->execute();
+    return $query->fetchAll();
+}
```

Comment:
I'd suggest splitting this into two functions actually, one function that creates the new PDO connection, sets the fetch mode and then returns $db

And then the second one that just does the query, execute and then returns the fetch all.

This way they are both more re-usable.